tests/irdl/test_declarative_assembly_format.py

```diff
@@ -2041,3 +2041,79 @@ class MixedOp(IRDLOperation):
 
     check_roundtrip(program, ctx)
     check_equivalence(program, generic_program, ctx)
+
+
+@irdl_op_definition
+class DefaultOp(IRDLOperation):
+    name = "test.default"
+
+    def_prop = prop_def(BoolAttr, default_value=BoolAttr.from_bool(False))
+
+    def_opt_prop = opt_prop_def(BoolAttr, default_value=BoolAttr.from_bool(True))
+
+    assembly_format = "(`prop` $def_prop^)? (`opt_prop` $def_opt_prop^)? attr-dict"
+
+
+@pytest.mark.parametrize(
+    "program, output, generic",
+    [
+        (
+            "test.default",
+            "test.default",
+            '"test.default"() <{"def_prop" = false}> : () -> ()',
+        ),
+        (
+            "test.default prop false opt_prop true",
+            "test.default",
+            '"test.default"() <{"def_prop" = false, "def_opt_prop" = true}> : () -> ()',
+        ),
+        (
+            '"test.default"() <{"def_prop" = false, "def_opt_prop" = true}> : () -> ()',
+            "test.default",
+            '"test.default"() <{"def_prop" = false, "def_opt_prop" = true}> : () -> ()',
+        ),
+        (
+            '"test.default"() <{"def_prop" = false}> : () -> ()',
+            "test.default",
+            '"test.default"() <{"def_prop" = false}> : () -> ()',
+        ),
+        (
+            "test.default prop true opt_prop false",
+            "test.default prop 1 opt_prop 0",
+            '"test.default"() <{"def_prop" = true, "def_opt_prop" = false}> : () -> ()',
+        ),
+    ],
+)
+def test_default_properties(program: str, output: str, generic: str):
+    ctx = MLContext()
+    ctx.load_op(DefaultOp)
+
+    parsed = Parser(ctx, program).parse_operation()
+    assert isinstance(parsed, DefaultOp)
+
+    stream = StringIO()
+    printer = Printer(stream=stream)
+    printer.print_op(parsed)
+
+    assert output == stream.getvalue()
+
+    stream = StringIO()
+    printer = Printer(stream=stream, print_generic_format=True)
+    printer.print_op(parsed)
+
+    assert generic == stream.getvalue()
+
+
+def test_default_accessors():
+    ctx = MLContext()
+    ctx.load_op(DefaultOp)
+
+    parsed = Parser(ctx, "test.default").parse_operation()
+
+    assert isinstance(parsed, DefaultOp)
+
+    assert parsed.def_prop.value.data == 0
+
+    assert parsed.def_opt_prop is not None
+
+    assert parsed.def_opt_prop.value.data == 1
```

Comment:
Can we also add a test here where we parse in the operation as a generic but don't specify in the dictionary an optioanl attribute/property? 

In MLIR the auto-generated getter (say `getDefOptProp`) should then still return `True`, while `getAttr("def_opt_propt")` would return null/missing